index.js

```diff
@@ -168,6 +168,18 @@ function session(options){
     // req.secret is passed from the cookie parser middleware
     var secrets = secret || [req.secret];
 
+    // Since we execute the functions (if they are a secet) we need
+    // to make a copy so that we don't overwrite the original function
+    var processedSecrets = [];
+
+    for (var i = 0; i < secrets.length; i++) {
+      if (typeof secrets[i] === 'function') {
+        processedSecrets.push(secrets[i](req));
```

Comment:
When invoking `secrets[i](req)`, some kind of validation should occur on the returned value and error out in some way if it's not a string.